assets/js/components/ResetButton.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * ResetButton component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Fragment, useState, useEffect, useCallback } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import {
+	clearWebStorage,
+	sanitizeHTML,
+} from '../util';
+import Dialog from './dialog';
+import Link from './Link';
+import Modal from './modal';
+import { STORE_NAME as CORE_SITE } from '../googlesitekit/datastore/site/constants';
+const { useSelect, useDispatch } = Data;
+
+function ResetButton( { children } ) {
+	const splashURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-splash' ) );
+
+	const [ dialogActive, setDialogActive ] = useState( false );
+	const postResetURL = addQueryArgs( splashURL, { notification: 'reset_success' } );
+
+	const { reset: resetCoreSite } = useDispatch( CORE_SITE );
+
+	const handleCloseModal = useCallback( ( event ) => {
+		if ( 27 === event.keyCode ) {
+			setDialogActive( false );
+		}
+	}, [ dialogActive ] );
+
+	useEffect( () => {
+		global.addEventListener( 'keyup', handleCloseModal, false );
+
+		return () => global.removeEventListener( 'keyup', handleCloseModal );
+	}, [ handleCloseModal ] );
+
+	const handleUnlinkConfirm = async () => {
+		resetCoreSite();
+		clearWebStorage();
+		handleDialog();
```

Comment:
I think the right UX will be to show a loader in place of reset button until the page redirects. What do you think?
Otherwise we can just keep on showing the dialog and redirect the page.

Reply:
I had a look and the existing UX definitely just closes the reset dialog and then redirects. It happens pretty quickly on a decent internet connection, but it isn't great.

I don't think showing a generic loading animation is a good indication of what's _currently happening_, because it would remove "Reset Site Kit" with "…" (some loading animation) so you might not _know_ what's loading/resetting. Changing the button to be disabled with the text "Site Kit is resetting" would be nice, but it's a UX change unrelated to the original IB and too much to add to this issue.

I think it'd be nicer if the dialog didn't _close_ on `handleUnlinkConfirm` though. But I'll file a follow-up issue to discuss all of this; for now let's leave it as-is 😄

Reply:
Filed #2347 to discuss.